tskit/trees.py

```diff
@@ -615,7 +615,12 @@ def is_sample(self, u):
     @property
     def num_nodes(self):
         """
-        Returns the number of nodes in the sparse tree.
+        Returns the number of nodes in the treeSequence that encapsulates this tree.
+        Equivalent to: ``tree.tree_sequence.num_nodes``.
+
+        note that a simplified SparseTree is a proper binary tree. meaning the
```

Comment:
Looks good, thanks @jgallowa07.

This second bit isn't true though --- we can have trees that aren't binary, even if they're simplified.

Reply:
Ah, then the best way to know all nodes associated with a single tree is simply `len([n for n in st.nodes()])`?

Reply:
Yep, ``len(list(tree.nodes()))`` is probably the best way to do it. We could add a method to compute it more efficiently if it's a common operation. How are you using it?